### PR TITLE
Expose missing spell checking interface

### PR DESCRIPTION
**Pull request by hferreiro**
This should fix https://github.com/brave/browser-laptop/issues/10849. Tested on Linux and Windows.

### CODE REVIEW

File: brave/browser/brave_content_browser_client.cc [hunk 1]
```diff
@@ -234,6 +234,15 @@ void BraveContentBrowserClient::ExposeInterfacesToRenderer(
   // gap between the preparation and the execution of that IPC.
   associated_registry->AddInterface(
       base::Bind(&CacheStatsRecorder::Create, render_process_host->GetID()));
+
+  scoped_refptr<base::SingleThreadTaskRunner> ui_task_runner =
+      content::BrowserThread::GetTaskRunnerForThread(
+          content::BrowserThread::UI);
+#if BUILDFLAG(ENABLE_SPELLCHECK)
+  registry->AddInterface(
```
Comment (bridiver) [inline brave/browser/brave_content_browser_client.cc]: that's odd, I noticed this when reviewing and I thought I saw it in the final PR

Comment (darkdh) [reply brave/browser/brave_content_browser_client.cc]: you are still missing 
```
+#if BUILDFLAG(HAS_SPELLCHECK_PANEL)
+    registry->AddInterface(base::Bind(&SpellCheckPanelHostImpl::Create),
+                           ui_task_runner);
+#endif
```
see 08fdafe21e81711142065b8251af2e5ecd268707

Comment (bridiver) [reply brave/browser/brave_content_browser_client.cc]: what is `HAS_SPELLCHECK_PANEL` ? are you sure that's something we need?

Comment (darkdh) [reply brave/browser/brave_content_browser_client.cc]: It is an [MacOS feature](https://support.apple.com/kb/PH18451?locale=en_US) and chromium uses it as a spell-correction aid for contentEditable, <input>
and <textarea>, and designMode document content that has the
focus on the current web page.
We new a instance of `SpellCheckPanel` in our `BraveContentRendererClient`

File: brave/browser/brave_content_browser_client.cc [hunk 2]
```diff
@@ -243,6 +243,10 @@ void BraveContentBrowserClient::ExposeInterfacesToRenderer(
       base::Bind(&SpellCheckHostImpl::Create, render_process_host->GetID()),
       ui_task_runner);
 #endif
+#if BUILDFLAG(HAS_SPELLCHECK_PANEL)
```
Comment (hferreiro) [inline brave/browser/brave_content_browser_client.cc]: @bridiver I believe this should be enclosed in the previous define (ENABLE_SPELLCHECK).

**Pull request merged**